### DISCUSSION TITLE
Daten bleiben aktiv nach Abschaltung eines WR

### DISCUSSION THREAD

**Discussion by fuxe13** [tbnobody/OpenDTU#498] [Q&A]
Hallo!
Ich habe die Verbindung der openDTU zum Victron Cerbo hergestellt. Einen Wechselrichter benutze ich für die Nachteinspeissung.
Dieser wird per Relais ein/aus geschalten. Wenn der WR in der Nacht abgeschalten wird, bleiben die Daten in der openDTU aktiv und werden so auch an den Cerbo für ein paar Stunden weitergeleitet. Wie/was könnte man einstellen damit das nach maximal 15-30 Minuten auf 0 Watt übergeben wird? Oder generell nach Verbindungsabbruch von (5-30Minuten) die Werte zurückgesetzt werden??

Würde mich über eine Lösung freuen
Walter

**Comment by dAjaY85** [comment]
Moin, es gibt die variable WR Reachable, diese geht auf 0 sobald der WR nicht mehr erreichbar ist.
Hab bei mir mit MQTT es so gemacht, sobald diese variable auf 0 geht, Poster HA auf allen MQTT topics eine null rein, süße bei den yield werten, diese werden um Mitternacht zurück gesetzt.

**Comment by fuxe13** [comment]
Hallo!

Das bedeutet?
Ich müsste mqqt aktivieren und dann irgendwie die Daten manipulieren dass ich am victron richtige Werte bekomme?
Hmmmmmm.......

----- Ursprüngliche Nachricht -----
Von: Andreas E. ***@***.***>
Datum: 08.01.2023 11:33
An: tbnobody/OpenDTU ***@***.***>
Kopie: fuxe13 ***@***.***>, Author ***@***.***>

Betreff: Re: [tbnobody/OpenDTU] Daten bleiben aktiv nach Abschaltung eines WR (Discussion #498)

**Reply by tbnobody** [reply]
mqtt sollte ja schon aktiv sein. ansonsten bekommst du die daten ja sowieso nicht in ein anderes system. oder verwendest du die web api? auch hier gibt es die entsprechenden felder.

**Comment by fuxe13** [comment]
Hmm...lauter Neuland....
Mqtt ist deaktiv und die Daten gelangen ohne Probleme in die Victron Cerbo.

**Reply by speedy1980** [reply]
Vermutlich mit diesem Projekt? https://github.com/henne49/dbus-opendtu
Da braucht es keinen MQTT. Cerbo / Venus holen sich die Daten über die API.

**Comment by fuxe13** [comment]
Re: [tbnobody/OpenDTU] Daten bleiben aktiv nach Abschaltung eines WR (Discussion #498)


Guten Tag speedy1980,

stimmt :)

aber was halt stört, dass wenn der WR rasch auf null geht (Nachteinspeissung) bleibt der allte export stehen und hab dadurch eine falsche anzeige.



am Mittwoch, 25. Jänner 2023 um 23:33 schrieben Sie:





Vermutlich mit diesem Projekt? https://github.com/henne49/dbus-opendtu
Da braucht es keinen MQTT. Cerbo / Venus holen sich die Daten über die API.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>





-- 
Mit freundlichen Grüßen
Walter Klammer
***@***.***

**Comment by benedictjaeger** [comment]
Ich habe ebenfalls das Problem. Mein HM-1500 verliert Abends die Verbindung und der letzte Wert bleibt stehen. Wie bei dir auch, liest mein Multiplus die OpenDTU über die API aus.  Resultat ist, dass meine PV für die Auswertung auch nachts weiter produziert. 

![image](https://user-images.githubusercontent.com/78529657/215342086-95855013-9e55-4db6-b68b-329511075b6b.png)

**Reply by Solarteur** [reply]
Ich verstehe die oben angesprochenen Probleme. Aber es gibt auch Nutzer, die keine weitere Auswertung machen und sich freuen, wenn die "Tagesdaten" bis zur erneuten Verbindung mit dem WR verfügbar sind.

**Reply by benedictjaeger** [reply]
Es geht auch nicht um die Tagesdaten, die können und sollen so bleiben. Es geht nur um die Momentanleistung. Wenn der WR nicht erreichbar ist, bringt einem die Watt Angabe keinen Mehrwert.

Worin liegt eigentlich der Unterschied zwischen dem X (Rot) und Ausrufezeichen (Gelb)?

**Reply by tbnobody** [reply]
Gelb == Leistung 0W
Rot == Keine Antwort

Aber man könnte im Python code einfach die aktuelle Leistung mit dem Reachable Wert multiplizieren. Dann wäre der Wert automatisch 0 wenn der Inverter nicht erreichbar ist.

**Reply by dAjaY85** [reply]
Kann man es nicht, ähnlich wie in ahoy, über die WebVisu an oder abwählen können, ob der Wert bei nicht Reachable auf 0 springt.
Nicht jeder hat dahinter eine Möglichkeit den Wert zu manipulieren.

**Reply by benedictjaeger** [reply]
> Gelb == Leistung 0W Rot == Keine Antwort
> 
> Aber man könnte im Python code einfach die aktuelle Leistung mit dem Reachable Wert multiplizieren. Dann wäre der Wert automatisch 0 wenn der Inverter nicht erreichbar ist.

Im HA kein Thema aber wer das Monitoring bzw. Auswertung im Victron / VRM; Portal nutzt, kann den Wert nicht mit dem Reachable Flag manipulieren

**Reply by tbnobody** [reply]
Warum nicht? Es muss ja eine Software geben die die Werte aus OpenDTU liest in Victron schreibt. Diese Software kann ja sehr wohl Werte manipulieren

**Reply by dAjaY85** [reply]
Victron liest es selber aus. Leider ist es eher closed.
Siehe die letzten Minuten hier:
https://youtu.be/PpjCz33pGkk

**Reply by tbnobody** [reply]
Woraus liest Victron es aus? Die opendtu-dbus Software kann ja sehr wohl Daten manipulieren und so anpassen das es für Victron passt.

**Reply by dAjaY85** [reply]
Was spricht den dagegen, in der WebVisu das anwählbar zu machen?

**Reply by tbnobody** [reply]
Das wird halt definitv nicht bald passieren weil ich aktuell den MQTT Teil usw. etwas umbaue. In Python ist es halt eine Zeile oder so.

**Reply by dAjaY85** [reply]
Ich kann's mal anschauen, sollte nicht so kompliziert sein.

**Reply by speedy1980** [reply]
Ich habe es mal angeschaut: https://github.com/henne49/dbus-opendtu/blob/main/dbus-opendtu.py
Zeile 350, da muss man ein if .. else bauen mit der Zeile:
 power = meter_data['inverters'][self.pvinverternumber]['0']['Power']['v']
Etwas in der Art:
if meter_data['inverters'][self.pvinverternumber]['producing']:
  power = meter_data['inverters'][self.pvinverternumber]['0']['Power']['v']
else:
  power = 0
Wenn ich es die nächsten Tage schaffe baue ich das mal manuell bei mir ein. 

Edit: Ich war mal so frei https://github.com/henne49/dbus-opendtu/issues/25

**Reply by dAjaY85** [reply]
Producing klappt nicht, brauchst Reachable.

**Reply by tbnobody** [reply]
Einfachster Fall wäre doch dann:
```python
power = meter_data['inverters'][self.pvinverternumber]['0']['Power']['v'] * meter_data['inverters'][self.pvinverternumber]['reachable']
```

**Reply by benedictjaeger** [reply]
> Einfachster Fall wäre doch dann:
> 
> ```python
> power = meter_data['inverters'][self.pvinverternumber]['0']['Power']['v'] * meter_data['inverters'][self.pvinverternumber]['reachable']
> ```

Habe den Einzeiler implementiert. Scheint zu funktionieren. Einfacher als gedacht. Danke

**Comment by fuxe13** [comment]
auf alle f&auml;lle w&auml;re das super, wenn das eun/umgebaut werden k&ouml;nnte!
  sag schon mal danke
  
  ----- Urspr&uuml;ngliche Nachricht -----
  Von: speedy1980 ***@***.***>
  Datum: 29.01.2023 23:38
  An: tbnobody/OpenDTU ***@***.***>
  Kopie: fuxe13 ***@***.***>, Author ***@***.***>
  Betreff: Re: [tbnobody/OpenDTU] Daten bleiben aktiv nach Abschaltung eines WR (Discussion #498)
  &nbsp; >

**Comment by fuxe13** [comment]
Leider wurde dieser kleine Bug noch nicht behoben, bzw. gib es noch keine besserung....

**Reply by benedictjaeger** [reply]
Der kleine Bug wurde in der DBUS Anbindung im Victron kompensiert.

**Comment by speedy1980** [comment]
Ist im dbus-opendtu behoben / verbessert.
https://github.com/henne49/dbus-opendtu/commit/663fa84c198eacb0aaad5875061b2712cad99461

**Reply by mhin76** [reply]
Ist bei mir leider auch so. Mit neuster Firmware. Wenn unter dem Tag ein HM zwei Mal nicht erreichbar ist, wird der Power Wert auf Null gesetzt (habe Nulle Laufzeit Daten auf Null gesetzt). Aber wenn er am Abend abschaltet, steht der letzte Wert bis morgens im OpenDTU beim HM drin. Doch ein Bug?

**Comment by fuxe13** [comment]
Guten Tag Benedict,

am Freitag, 17. Februar 2023 um 17:36 schrieben Sie:

> Der kleine Bug wurde in der DBUS Anbindung im Victron kompensiert.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.
vielen dank für die antwort :)

-- 
Mit freundlichen Grüßen
Walter Klammer
***@***.***

**Comment by fuxe13** [comment]
Vielen Dank....aber
 
wäre es Möglich für mich eine Idiotensichere Updatebeschreibung zu machen und das in Deutsch.... 
für euch ist es logisch und einfach, aber für mich..naja. Habs auch nur geschafft, da "meine Energiewende" es schritt für schritt gezeigt hatte...
 
sag schon mal: recht herzlichen DANK
Walter
 
 

am Freitag, 17. Februar 2023 um 23:03 schrieben Sie:

> Ist im dbus-opendtu behoben / verbessert.
> ***@***.***
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.

-- 
Mit freundlichen Grüßen
Walter Klammer
***@***.***

**Comment by fuxe13** [comment]
die version für die opendtu Firmware Version / Git Hash | ee17dc8 funktioniert der datenexport.
alle neueren versionen darüber werden keine daten an die victron cerbo weitergegeben.

und wie man an der cerbo das update einspielen kann, steh ich auch an.....

**Comment by mhin76** [comment]
> Hallo!
> 
> Ich habe die Verbindung der openDTU zum Victron Cerbo hergestellt. Einen Wechselrichter benutze ich für die Nachteinspeissung.
> 
> Dieser wird per Relais ein/aus geschalten. Wenn der WR in der Nacht abgeschalten wird, bleiben die Daten in der openDTU aktiv und werden so auch an den Cerbo für ein paar Stunden weitergeleitet. Wie/was könnte man einstellen damit das nach maximal 15-30 Minuten auf 0 Watt übergeben wird? Oder generell nach Verbindungsabbruch von (5-30Minuten) die Werte zurückgesetzt werden??
> 
> 
> 
> Würde mich über eine Lösung freuen
> 
> Walter

Hallo Fuxe

Sehr interessant dein Setup. Du hast also einen Akku vor einem HM platziert? Wie lädst du den auf? Wie viel Volt hat er und wie steuerst du bzgl wieviel er umsetzen soll?

**Reply by fuxe13** [reply]
Hallo mhin76,

Lade die Akkus über eine Inselanlage, welche auch als Notstrom geschalten werden kann. Wir haben ein 48v System. Als Begrenzer dient ein Victron MPPT 75/15. Somit kommen am Laderegler 24vx15a an. Das genügt für den MH300. Natürlich sind schon einige Verluste vorhanden. Läuft aber zuverlässig.
